core/src/main/scala/org/apache/spark/sql/SnappyStrategies.scala

```diff
@@ -61,17 +61,35 @@ private[sql] trait SnappyStrategies {
   object LocalJoinStrategies extends Strategy {
     def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
       case ExtractEquiJoinKeys(joinType, leftKeys, rightKeys, condition,
-      left, right) =>
-        if (canBuildRight(joinType) && canLocalJoin(right)) {
-          makeLocalHashJoin(leftKeys, rightKeys, left, right, condition,
-            joinType, joins.BuildRight)
-        } else if (canBuildLeft(joinType) && canLocalJoin(left)) {
+      left, right) if canBuildRight(joinType) && canLocalJoin(right) =>
+        makeLocalHashJoin(leftKeys, rightKeys, left, right, condition,
+          joinType, joins.BuildRight, true)
+      case ExtractEquiJoinKeys(joinType, leftKeys, rightKeys, condition,
+      left, right) if canBuildLeft(joinType) && canLocalJoin(left) =>
           makeLocalHashJoin(leftKeys, rightKeys, left, right, condition,
-            joinType, joins.BuildLeft)
-        } else Nil
+            joinType, joins.BuildLeft, true)
+      case ExtractEquiJoinKeys(joinType, leftKeys, rightKeys, condition, left, right)
+        if canBuildRight(joinType) && canBuildLocalHashMap(right) ||
+            !RowOrdering.isOrderable(leftKeys) =>
```

Comment:
RowOrdering  checks if the keys can be sorted. For e.g. user objects may not be orderable. In such cases, we can only do hash joins. However, this check is picked up from SparkStrategies as is and its utility for us is debatable.